torch/accelerator/__init__.py

```diff
@@ -0,0 +1,132 @@
+# mypy: allow-untyped-defs
+r"""
+This package introduces support for the current :ref:`accelerator<accelerators>` in python.
+"""
+
+from typing import Any
+
+import torch
+
+from ._utils import _device_t, _get_device_index
+
+
+def current_accelerator() -> torch.device:
+    r"""Return the device type of the current :ref:`accelerator<accelerators>`.
+
+    Returns:
+        torch.device: return the current accelerator as :class:`torch.device`.
+            If no available accelerators, return cpu device.
+
+    Example::
+
+        >>> # xdoctest: +SKIP
+        >>> if torch.accelerator.current_accelerator().type == 'cuda':
+        >>>     is_half_supported = torch.cuda.has_half()
+        >>> elif torch.accelerator.current_accelerator().type == 'xpu':
+        >>>     is_half_supported = torch.xpu.get_device_properties().has_fp16
+    """
+    return torch._C._accelerator_getAccelerator()
+
+
+def device_count() -> int:
+    r"""Return the number of current :ref:`accelerator<accelerators>` available.
+
+    Returns:
+        int: the number of the current :ref:`accelerator<accelerators>` available.
+            If no available accelerators, return 0.
+    """
+    return torch._C._accelerator_deviceCount()
+
+
+def is_available() -> bool:
+    r"""Check if there is an available :ref:`accelerator<accelerators>`.
+
+    Returns:
+        bool: A boolean indicating if there is an available :ref:`accelerator<accelerators>`.
+
+    Example::
+
+        >>> assert torch.accelerator.is_available() "No available accelerators detected."
+    """
+    return device_count() > 0
+
+
+def current_device_idx() -> int:
+    r"""Return the index of a currently selected device for the current :ref:`accelerator<accelerators>`.
+
+    Returns:
+        int: the index of a currently selected device.
+    """
+    return torch._C._accelerator_getDeviceIndex()
+
+
+def set_device(device: _device_t) -> None:
```

Comment:
Should this one be set_device_idx() actually to match the getter above?

Reply:
Paired with `current_device_idx`, `set_device_idx` looks more reasonable.
Similar to the BC issue of `current_device_idx,` if the user confuses the API name change, an alias name `set_device` should be given.